sim/midas/src/main/cc/emul/context.h

```diff
@@ -0,0 +1,59 @@
+#ifndef _EMUL_CONTEXT_H
+#define _EMUL_CONTEXT_H
+// clang-format off
+
+// This was copied from riscv-fesvr, which is now hosted in spike.
+// Disable formatting to minimize the diff
+
+// A replacement for ucontext.h, which is sadly deprecated.
+
+#include <pthread.h>
+
+#if defined(__GLIBC__)
+# undef USE_UCONTEXT
+# define USE_UCONTEXT
+# include <ucontext.h>
+# include <memory>
+#include <limits.h>
+
+#if (ULONG_MAX > UINT_MAX) // 64-bit systems only
+#if (100*GLIB_MAJOR_VERSION+GLIB_MINOR_VERSION < 208)
+#define GLIBC_64BIT_PTR_BUG
+static_assert (sizeof(unsigned int)  == 4, "uint size doesn't match expected 32bit");
```

Comment:
What is the source of this problem? Why noi use `uint32_t` for 4-byte values, `uint64_t` for 8-byte values and `uintptr_t` for pointers cast to integers?

Reply:
I'm not entirely sure what the context was, but i'd rather not fiddle too much with this code.